components/actions/blocks/action-dependencies/ActionDependenciesBlock.tsx

```diff
@@ -22,16 +27,45 @@ type ActionGroup = {
 };
 
 type Props = {
+  action: Action;
   activeActionId?: string;
   size?: 'default' | 'small';
-  actionGroups: ActionGroup[];
   showTitle?: boolean;
   title?: string;
   helpText?: string;
+  getFullAction: (id: string) => Action;
+  skeleton?: boolean;
```

Comment:
Nitpick: Consider renaming `skeleton` to `loading` to reflect the state rather than its visual representation